packages/beacon-node/src/api/impl/beacon/pool/index.ts

```diff
@@ -27,6 +27,17 @@ export function getBeaconPoolApi({
     async getPoolAttestations({slot, committeeIndex}) {
       // Already filtered by slot
       let attestations = chain.aggregatedAttestationPool.getAll(slot);
+
+      if (committeeIndex !== undefined) {
+        attestations = attestations.filter((attestation) => committeeIndex === attestation.data.index);
+      }
+
+      return {data: attestations};
+    },
+
+    async getPoolAttestationsV2({slot, committeeIndex}) {
+      // Already filtered by slot
+      let attestations = chain.aggregatedAttestationPool.getAll(slot);
       const fork = chain.config.getForkName(slot ?? attestations[0].data.slot) ?? ForkName.phase0;
```

Comment:
This logic is unsafe and also incorrect, need to be careful with accessing items in array as typescript does not properly warn here

If array is empty, this will give an error `attestations[0].data.slot`, need to do `attestations[0]?.data.slot`

I would also favor a logic like this

```ts
const fork = chain.config.getForkName(slot ?? attestations[0]?.data.slot ?? chain.clock.currentSlot);
```

Reply:
Honestly what we should do across the code base is type all arrays like this

```ts
(Attestation | undefined)[]
```

instead of 

```ts
Attestation[]
```